scripts/cluster-upgrade.sh

```diff
@@ -310,18 +310,24 @@ upgrade() {
     log $OCM_NAME "upgrade" "ClusterVersion on $TO"
     log $OCM_NAME "upgrade" "checking node versions"
 
-    # fun fact!  after clusterversion says it is done, individual nodes could still be updated.
-    MACHINE_COUNT_ALL=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get nodes --no-headers | wc -l)
+    MASTER_MACHINEPOOL_TOTAL_COUNT=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get machineconfigpools -o json  | jq -r ".items[] | select(.metadata.name == \"master\" ) | .status.machineCount")
+    MASTER_MACHINEPOOL_UPDATED_COUNT=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get machineconfigpools -o json  | jq -r ".items[] | select(.metadata.name == \"master\" ) | .status.updatedMachineCount")
```

Comment:
I know we're not super concerned about performance and resource consumption here, but why make two API calls when one will do?

Also, using `-o jsonpath` rather than `-o json | jq` not only saves us some subprocesses, but does all the filtering on the server side, saving network traffic. (Again, in this case the amount of data is tiny, so not really an issue.)

```
DATA=($(oc get machineconfigpools -o jsonpath="{.items[?(@.metadata.name=="'"master"'")]['.status.machineCount', '.status.updatedMachineCount']}"))
MASTER_MACHINEPOOL_TOTAL_COUNT=${DATA[0]}
MASTER_MACHINEPOOL_UPDATED_COUNT=${DATA[1]}
```

kind of thing.

Not suggesting a change here, really, just musing.

Reply:
Here, we are computing the upgradecount and total count separately and monitor in a loop. That would work no doubt. But I was even thinking along the lines of fetching the status directly from machineconfigpool. Something like:

`oc get mcp -ojson | jq -r '.items[].status.conditions[] | select(.type == "Updated" and .status == "True")`

So we can check for all mcp that are configured and check for their "Updated"="True" status. This way, even if there is any separate mcp added (Eg. for infra), we can still get the work done with same logic. Plus, it required less lines of script too.

Reply:
Could but then have to add the various values together.  Given it's already written to handle master and worker separate I'm in favor of keeping it that way.  Other option is drop master and simply keep the worker check but it's not really critical to tweak that.  I like having a fail-safe to check master node upgrades.